kubernetes/k8s-gencfg

```diff
@@ -12,7 +12,15 @@ def parse_file(args, input_file, output_file):
     with open(input_file) as file:
         input_data = file.read()
 
-    output_data = input_data.replace("{{IMAGE_PATH}}", args.image_path).replace("{{IMAGE_TAG}}", args.image_tag)
+    data = input_data.replace("{{IMAGE_PATH}}", args.image_path)
+    data = data.replace("{{IMAGE_TAG}}", args.image_tag)
+
+    # this argument is optional
+    # ~ means null in yaml
+    output_data = data.replace("{{IMAGE_SECRET}}", args.image_secret if (args.image_secret != None) else "~")
```

Comment:
I'll probably try this out a bit on our rigs, to make sure it behaves as i'm expecting, but can you give a summary of any testing/checking you've done here on if there isn't a secret provided?

Reply:
I haven't done proper testing of the no secret case, because I don't have a public repo. I did try running it on my private repo, and it seemed to parse fine and then give an authentication failed error.

Today I created a public docker hub repo. I'll try on that when I get a chance.

Reply:
Yes it does work with no authentication. I just tried.